enhancements/cguConditionEnhancement.md

```diff
@@ -0,0 +1,452 @@
+
+ ```yaml
+  Type:    "PreUpgradeValidation",
+  Status:  True/False,
+  Reason:  "MissingManagedPolicy/InvalidPlatformImage/ValidationComplete",
+  Message: “”
+  ```
+
+4. Condition type: **PrecacheSpecValid** and **PrecachingDone** which is already in the current design. For **PrecacheSpecValid**, we may omit the highlighted Reason from the [current spec](#precachingSpec)
+5. Checking the blocking CRs status, if there is any issue UpgradeCannotStart reason is reported. If CRs are not blocked the upgrade can start. Also, adding and deleting labels to the cluster can be performed here. Condition type: Started/Progressing?
+
+ ```yaml
+    Type:    "Started/Progressing",
+    Status:  True/False,
+    Reason:  "MissingBlockingCR/IncompleteBlockingCR/MissingClusterLabel/InProgress",
+    Message: “”
+  ```
+
+  > Mohamed had a comment on this block: "In my opinion those are 2 different conditions 1) Initialized/Started means that the action (copying policies) is happening  2) Progressing/InProgress means that policy remediation started." To discuss.
+
+  6. The **UpgradeCompleted** and **UpgradeTimedOut** Reason can be moved to condition type: Available/Ready
+
+  ```yaml
+    Type:    "Available/Ready",
+    Status:  True/False,
+    Reason:  "UpgradeCompleted/UpgradeTimedOut",
+    Message: “”
+  ``` 
+
+### Workflow summary
+
+- When backup and Precaching is true:
+  * ClusterValidation/ClusterSelected -> BackupDone -> PreUpgradeValidation -> PrecacheSpecValid -> PrecachingDone -> Started/Progressing -> Available/Ready
+ 
+
+- When backup and precaching is false:
+  * ClusterValidation/ClusterSelected -> PreUpgradeValidation -> Started/Progressing -> Available/Ready
+
+### Mapping between Type, Status, Reason and Message:
+
+if backup and/or precaching is false, the related types will not appear in `.status.conditions`.
+
+
+  | Type | Status| Reason| Message |
+  |------|-------|---------|--------|
+  ClusterValidation/ClustersSelected| True| Completed||
+  | |False | NotFound/NotPresent| |
+  `BackupDone` | True | BackupCompleted | Backup is completed |
+  | | False | BackupNotDone | Backup is required and not done |
+  PreUpgradeValidation| True | Completed||
+  | | False | MissingManagedPolicy| |
+  | | False | InvalidPlatformImage| | 
+  `PrecacheSpecValid` | True | PrecacheSpecIsWellFormed | Pre-caching spec is valid and consistent|
+  | | False | NotAllManagedPoliciesExist| The ClusterGroupUpgrade CR has managed policies that are missing|
+  `PrecachingDone` | True | Completed | Precaching is completed |
+  | | False | PrecachingNotDone | Precaching is required and not done | 
+  Started/Progressing| True | InProgress||
+  | | False | MissingBlockingCR | |
+  | | False | IncompleteBlockingCR | | 
+  | | False | MissingClusterLabel | | 
+  Available/Ready/Completed| True | UpgradeCompleted||
+  | | False | UpgradeTimedOut | |
```

Comment:
I guess we should also have upgrade failed condition and reasons could be Timeout /Failed to connect/Any other reason

Reply:
For me, Available/Ready/Completed -> any of this is enough to indicate if the upgrade has failed or succeeded. Failed/timedout can be in the reason with status false.